src/TraceEvent/Computers/TraceProcess.cs

```diff
@@ -27,24 +27,27 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceProcessesExtensions
     {
+        internal static readonly string ProcessesKey = "Computers/Processes";
+
         public static void NeedProcesses(this TraceEventDispatcher source)
         {
             TraceProcesses processes = source.Processes();
-            if (processes == null || m_currentSource != source)
+            if (processes == null)
             {
```

Comment:
➡️ In the pull request as it currently stands, it is no longer possible for the callbacks to be registered multiple times. In the past, it was possible under concurrent usage scenarios for the callbacks to be registered multiple times (frighteningly, we're talking about two independent sections of code processing two independent traces here, with no shared objects).